pkg/controller/controller.go

```diff
@@ -97,18 +97,34 @@ func (c *Controller) initOperatorConfig() {
 		c.logger.Infoln("no ConfigMap specified. Loading default values")
 	}
 
-	// env var takes priority over the same param from the operator ConfigMap
-	watchedNamespace := os.Getenv("WATCHED_NAMESPACE")
-	if watchedNamespace != "" {
-		c.logger.Infof("Watch the %q namespace specified in the env variable WATCHED_NAMESPACE\n", watchedNamespace)
-		configMapData["watched_namespace"] = watchedNamespace
-	}
+	// by default, the operator listens to all namespaces
+	// by setting the env variable, one can restrict the operator to a single namespace
+	watchedNamespace, isPresentInEnv := os.LookupEnv("WATCHED_NAMESPACE")
+	if isPresentInEnv {
+		// special case: v1.NamespaceAll currently also evaluates to the empty string
+		// so when the env var is set to the empty string, use the default ns
+		// since the meaning of this env var is only one namespace
+		if watchedNamespace == "" {
+			c.logger.Infof("The WATCHED_NAMESPACE env var evaluates to the empty string, falling back to watching the 'default' namespace.\n", watchedNamespace)
```

Comment:
It feels little bit too implicit for me. Maybe it would be better if:
* when this variable is not present, listen to a default namespace (which would be close to current behavior)
* when this variable is there and not empty, listen to these namespaces
* when this variable is there and empty, throw a validation error
* add a special flag "watchAll" to watch all namespaces.